packages/react/src/feature-app-container.tsx

```diff
@@ -162,8 +162,17 @@ type InternalFeatureAppContainerProps<
   Pick<FeatureHubContextConsumerValue, 'featureAppManager' | 'logger'>;
 
 type InternalFeatureAppContainerState<TFeatureApp extends FeatureApp> =
-  | {readonly featureAppError: Error}
-  | {readonly featureApp: TFeatureApp; readonly loading: boolean};
+  | {
+      readonly error: Error;
+      readonly featureApp?: TFeatureApp;
+      readonly failedToHandleAsyncError?: boolean;
+      readonly loading: boolean;
+    }
+  | {
+      readonly featureApp: TFeatureApp;
+      readonly failedToHandleAsyncError?: boolean;
+      readonly loading: boolean;
+    };
```

Comment:
While we talked about removing the Union type here, I found that it is actually quite helpful to have one "working" and one "errored" state type, as it reflects the component's logic and the checks. removing the union would mean I'd have lots of untested boolean branches that never happen.

However, the previous type was just wrong, so I fixed it to actually reflect the react state, that only gets merged into by setStates.